index.html

```diff
@@ -3750,6 +3777,180 @@ <h2>
+            <var>base</var>.
+            </li>
+            <li>If <var>validationURL</var> is failure, throw a
+            <a>TypeError</a>.
+            </li>
+            <li>Let <var>event</var> be the result of <a data-cite=
+            "DOM#concept-event-constructor">constructing</a> a
+            <a>MerchantValidationEvent</a> instance with <var>type</var> and
+            <var>eventInitDict</var>.
+            </li>
+            <li>Set <var>event</var>.<a>[[\waitForUpdate]]</a> to false.
+            </li>
+            <li>Set <var>event</var>.<a>[[\validationURL]]</a> to
+            <var>validationURL</var>.
+            </li>
+            <li>Return <var>event</var>.
+            </li>
+          </ol>
+        </section>
+        <section>
+          <h3>
+            <dfn>validationURL</dfn> attribute
+          </h3>
+          <p>
+            A <a data-cite="url#concept-url">URL</a> from which a developer can
+            fetch <a>payment handler</a>-specific verification data. By then
+            passing that data (or a promise that resolves with that data) to
+            <a>complete()</a>, the user agent can verify that the payment
+            request is from a authorized merchant.
+          </p>
+          <p>
+            When getting, the user agent MUST return the <a>context
+            object</a>'s <a>[[\validationURL]]</a> internal slot.
+          </p>
+        </section>
+        <section>
+          <h3>
+            <dfn>complete()</dfn> method
+          </h3>
+          <p data-test=
+          "payment-request/MerchantValidationEvent/complete-method-manual.https.html">
+            The <a>MerchantValidationEvent</a>'s
+            <code>complete(<var>merchantSessionPromise</var>)</code> method
+            MUST act as follows:
+          </p>
+          <ol class="algorithm">
+            <li>Let <var>event</var> be the <a>context object</a>.
+            </li>
+            <li>If <var>event</var>'s <a>isTrusted</a> attribute is false, then
+            then <a>throw</a> an "<a>InvalidStateError</a>"
+            <a>DOMException</a>.
+            </li>
+            <li>Let <var>request</var> be <var>event</var>.<a data-lt=
+            "MVE.[[request]]">[[\request]]</a>.
+            </li>
+            <li>If <var>request</var>.<a>[[\state]]</a> is not
+            "<a>interactive</a>", then <a>throw</a> an
+            "<a>InvalidStateError</a>" <a>DOMException</a>.
+            </li>
+            <li>If <var>request</var>.<a>[[\updating]]</a> is true, then
+            <a>throw</a> an "<a>InvalidStateError</a>" <a>DOMException</a>.
+            </li>
+            <li>Set <var>event</var>'s <a>stop propagation flag</a> and <a>stop
+            immediate propagation flag</a>.
+            </li>
+            <li>Set <var>event</var>.<a>[[\waitForUpdate]]</a> to true.
+            </li>
+            <li>Run the <a>validate a merchant algorithm</a> with
```

Comment:
Wrong algorithm

Reply:
Oh crap. I went all circular (how embarrassing!)... this was supposed to:

 1.  disable the UI. 
 2. wait for wait for the promise to settle, "abort the update" is it rejects... 
 3. if it fulfills, then `value` needs to be cryptographically verified. 
 4. if it's the merchant is no good, abort the update.
 5. if merchant is validated, enable the UI.